reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClient.java

```diff
@@ -1105,14 +1105,23 @@ public final HttpClient mapConnect(
 	 * @since 0.9.7
 	 */
 	public final HttpClient metrics(boolean enable, Function<String, String> uriTagValue) {
+		return metrics(enable, () -> configuration().defaultMetricsRecorder(), uriTagValue);
+	}
+
+	@Override
+	public final HttpClient metrics(boolean enable, Supplier<? extends ChannelMetricsRecorder> recorder) {
+		return metrics(enable, recorder, configuration().uriTagValue());
+	}
+
+	public final HttpClient metrics(boolean enable, Supplier<? extends ChannelMetricsRecorder> recorder, Function<String, String> uriTagValue) {
 		if (enable) {
 			if (!Metrics.isInstrumentationAvailable()) {
 				throw new UnsupportedOperationException(
 						"To enable metrics, you must add the dependency `io.micrometer:micrometer-core`" +
 								" to the class path first");
 			}
```

Comment:
Do we need to check for Micrometer as this is a custom recorder that might not need it?
We need it only in case of the default implementation.

Reply:
I'll fix here as well